pydis_site/apps/api/models/bot/filters.py

```diff
@@ -0,0 +1,260 @@
+
+
+class FilterBase(ModelTimestampMixin, ModelReprMixin, models.Model):
+    """One specific trigger of a list."""
+
+    content = models.CharField(max_length=100, help_text="The definition of this filter.")
+    description = models.CharField(
+        max_length=200,
+        help_text="Why this filter has been added.", null=True
+    )
+    additional_field = models.JSONField(null=True, help_text="Implementation specific field.")
+    filter_list = models.ForeignKey(
+        FilterList, models.CASCADE, related_name="filters",
+        help_text="The filter list containing this filter."
+    )
+    dm_content = models.CharField(
+        max_length=1000,
+        null=True,
+        blank=True,
+        help_text="The DM to send to a user triggering this filter."
+    )
+    dm_embed = models.CharField(
+        max_length=2000,
+        help_text="The content of the DM embed",
+        null=True,
+        blank=True
+    )
+    infraction_type = models.CharField(
+        choices=[
+            (choices[0].upper(), choices[1])
+            for choices in [("NONE", "None"), *Infraction.TYPE_CHOICES]
+        ],
+        max_length=10,
+        null=True,
+        help_text="The infraction to apply to this user."
+    )
+    infraction_reason = models.CharField(
+        max_length=1000,
+        help_text="The reason to give for the infraction.",
+        blank=True,
+        null=True
+    )
+    infraction_duration = models.DurationField(
+        null=True,
+        help_text="The duration of the infraction. Null if permanent."
+    )
+    infraction_channel = models.BigIntegerField(
+        validators=(
+            MinValueValidator(
+                limit_value=0,
+                message="Channel IDs cannot be negative."
+            ),
+        ),
+        help_text="Channel in which to send the infraction.",
+        null=True
+    )
+    guild_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers.",
+        null=True
+    )
+    filter_dm = models.BooleanField(help_text="Whether DMs should be filtered.", null=True)
+    dm_pings = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Who to ping when this filter triggers on a DM.",
+        null=True
+    )
+    remove_context = models.BooleanField(
+        help_text=(
+            "Whether this filter should remove the context (such as a message) "
+            "triggering it."
+        ),
+        null=True
+    )
+    bypass_roles = ArrayField(
+        models.CharField(max_length=100),
+        help_text="Roles and users who can bypass this filter.",
+        null=True
+    )
+    enabled = models.BooleanField(
+        help_text="Whether this filter is currently enabled.",
+        null=True
+    )
+    send_alert = models.BooleanField(
+        help_text="Whether an alert should be sent.",
+        null=True
+    )
+
+    # Check FilterList model for information about these properties.
```

Comment:
I'm not sure I see anything in the `FilterList` model that isn't here? Help text here seems fine though, fine to leave exact specification to the bot side.

Reply:
This code has gone through so many iteration I have no clue why this was written. 4eecda9